contracts/Staking.sol

```diff
@@ -664,6 +689,35 @@ contract Staking is Pauser, Whitelist {
         emit DelegationUpdate(_valAddr, delAddr, validator.tokens, delegator.shares, -int256(_tokens));
     }
 
+    function _removeFrom(
+        address _addr,
+        address[] storage _addrArray,
+        int256 _index
+    ) private {
+        uint256 index = uint256(_index);
+        require(index == uint256(int256(-1)) || _addrArray[index] == _addr, "index not found");
+        uint256 lastIndex = _addrArray.length - 1;
+        if (index == uint256(int256(-1))) {
+            for (uint256 i = 0; i < _addrArray.length; i++) {
+                if (_addrArray[i] == _addr) {
+                    if (i < lastIndex) {
```

Comment:
A way to reduce the dup code is to set index = i (or replace i with index) and then break `if _addrArray[index] == _addr`.

Further, we could add a check just after the loop
`require(index <= lastIndex, "address not found")`

and then remove the `if` on line 711

Reply:
Fixed accordingly.